### PR TITLE
Debug stuff

### PR DESCRIPTION
**Pull request by c-wide**
**Pull Request Description**

This PR adds a boot controller that will perform various tasks/checks when the server resource starts and also adds a simple export check if PhoneAsItem is enabled.

This is WIP for a few reasons:

- Do you guys want this to be a controller/service or just a bunch of util functions I run in the server.ts file on startup?
- If it should be a controller/service, is "boot" the correct name? I think it accurately describes whats happening but I would like input.
- If boot is fine, should I remove the onServerResourceStart event handler from server.ts and provide confirmation after checks have been completed?
- After each check should I be providing confirmation that it passed or just log when an error occurs?
- How should the logs be formatted? Do we want a huge box like build-checker or just a simple colored log?
- Where should I put "requiredDbColumns" and "frameworkDependencies" that are located at the top of the boot service file or is this location fine?
- In cl_main I added an export checker if PhoneAsItem.enabled is true, should I make an event and also log the error to the server or is just client logging fine for this?

**Pull Request Checklist**:
* [x] Have you followed the guidelines in our Contributing document and Code of Conduct?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/project-error/npwd/pulls) for the same update/change?
* [x] Have you built and tested NPWD in-game after the relevant change?

### CODE REVIEW

File: resources/server/boot/boot.service.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import BootDb, { _BootDb } from './boot.db';
+import { bootLogger } from './boot.utils';
+import { config } from '../server';
+
+const { identifierColumn, phoneNumberColumn } = config.database;
+const requiredDbColumns = [identifierColumn, phoneNumberColumn];
+
+const frameworkDependencies = {
+  es_extended: ['esx-npwd'],
```
Comment (RockySouthpaw) [inline resources/server/boot/boot.service.ts:9]: Should this not be 'es_extended'

Comment (c-wide) [reply resources/server/boot/boot.service.ts:9]: The prettier config auto corrects it

Comment (RockySouthpaw) [reply resources/server/boot/boot.service.ts:9]: Well qb-core has '' so kind of weird es_extended wouldn’t

Comment (itschip) [reply resources/server/boot/boot.service.ts:9]: @RockySouthpaw It's because the `_` is valid in a object property, while `-` is not :/

**Comment by c-wide** [comment]
Requested changes made

**Comment by c-wide** [comment]
Should be good for review/testing and merge

**Review by itschip — APPROVED** [review]
Tested and seems to work fine. LGTM :)

**Pull request merged**